unittests/test_utility.py

```diff
@@ -236,18 +236,18 @@ def test_is_url():
 
 
 @pytest.fixture
-def no_git_installed():
+def git_only():
     try:
-        completed = osext.run_command(f'git --version', check=True, log=False)
+        completed = osext.run_command('git --version', check=True, log=False)
     except (SpawnedProcessError, FileNotFoundError):
-        return True
-
-    return False
+        pytest.skip('no git installation found on system')
 
 
-def test_git_repo_hash(monkeypatch, no_git_installed):
-    if no_git_installed:
-        pytest.skip('no git installation found on system')
+def test_git_repo_hash(monkeypatch, git_only):
+    try:
+        completed = osext.run_command('git status', check=True, log=False)
+    except (SpawnedProcessError, FileNotFoundError):
```

Comment:
Why don't you simply check if `.git` exists?

Reply:
I think it's better to rely on git to check if it's a valid repo.

Reply:
My only problem is whether this is worth it. This four lines of code and much more verbose instead of just two.

Reply:
`.git` can exist but in case it's not a proper repo you cannot get the hash.

Reply:
Ok, I see the scenario that sb downloads the source (without cloning), has git and runs the unit tests, so this check has a meaning. I'm not gonna argue further about the way for writing it.